src/CommonITILObject.php

```diff
@@ -6929,7 +6929,7 @@ public function getTimelineItemtypes(): array
                 if (is_callable($hook_itemtypes)) {
                     $hook_itemtypes = $hook_itemtypes(['item' => $this]);
                 }
-                $itemtypes = array_merge($itemtypes, $hook_itemtypes);
+                $itemtypes = array_merge($itemtypes, $hook_itemtypes ?? []);
```

Comment:
A warning could help plugin developpers to identify the issue.
```suggestion
                if (is_array($hook_itemtypes)) {
                    $itemtypes = array_merge($itemtypes, $hook_itemtypes);
                } else {
                    trigger_error(
                        sprintf('"%s" hook callback result should be an array, "%s" returned.', Hooks::TIMELINE_ANSWER_ACTIONS, gettype($hook_itemtypes)),
                        E_USER_WARNING
                    );
                }
```

Reply:
I don't think that will be very effective. It would be very clear what the issue is if the wrong value type is returned here. The issue is that plugin developers need to test cases that their hooks do not handle.

For example, for the One-time Secret plugin, they check if the itemtype is a Ticket and not solved and return an array with their action. Otherwise, they return nothing. If they tried viewing a solved/closed ticket or any change or problem, they should have seen the same issue as described in the linked issue.

The solution I've thought of so far is to provide one or more traits with all of the hook function definitions. Plugin developers can apply the traits to their existing classes as needed and override the hooks they want, or create a new class to centralize the hook code.

Reply:
> The solution I've thought of so far is to provide one or more traits with all of the hook function definitions. Plugin developers can apply the traits to their existing classes as needed and override the hooks they want, or create a new class to centralize the hook code.

We planned to do it in GLPI 10.0, but we do not had time for this.

But for the short time, with your proposal, if returned value is a string on a plugin, it will produce a fatal error, so we should still ensure that returned value matches expected type.
```Fatal error: Uncaught TypeError: array_merge(): Argument #1 must be of type array, string given```

Reply:
That seems overly complicated but this isn't the place for that discussion I guess.